src/server/enterprise/server/api_server.go

```diff
@@ -332,3 +346,237 @@ func (a *apiServer) Deactivate(ctx context.Context, req *ec.DeactivateRequest) (
 
 	return &ec.DeactivateResponse{}, nil
 }
+
+// Pause sets the cluster to paused mode, restarting all pachds in a paused
+// status.  If they are already paused, it is a no-op.
+func (a *apiServer) Pause(ctx context.Context, req *ec.PauseRequest) (resp *ec.PauseResponse, retErr error) {
+	if a.env.Mode == SidecarMode || a.env.Mode == EnterpriseMode {
```

Comment:
I think it may make more sense to make this an allow list as opposed to a disallow list. i.e. `if a.env.Mode != "full" then error`. This way if new modes are added in the future without knowledge of this feature, we protect them from being paused unless desired